### PR TITLE
feat(nfts): add in make offer ui logic

### PR DESCRIPTION
**Pull request by rsteubs-bc**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/components/Form/NumberBox/index.js [hunk 1]
```diff
@@ -36,6 +36,10 @@ const NumberBox = (field) => {
   return (
     <Container className={field.className}>
       <NumberInput
+        style={{
```
Comment (rsteubs-bc) [inline packages/blockchain-wallet-v4-frontend/src/components/Form/NumberBox/index.js:39]: To-Do: remove

File: packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/MakeOffer/index.tsx [hunk 2]
```diff
@@ -166,30 +189,85 @@ const MakeOffer: React.FC<Props> = (props) => {
...
+                    component={SelectBox}
+                    elements={[
+                      {
+                        group: '',
+                        items: map(
+                          (item) => ({
+                            text: item.text,
+                            value: item.value
+                          }),
+                          [
```
Comment (rsteubs-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/MakeOffer/index.tsx:243]: To-Do: list as read only

File: packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/MakeOffer/index.tsx [hunk 3]
```diff
@@ -166,30 +189,85 @@ const MakeOffer: React.FC<Props> = (props) => {
...
+                            { text: 'Network Fees', value: 'network' },
+                            { text: 'Offer Fees', value: 'offer' },
+                            { text: 'Wrapped Eth Fees', value: 'weth' }
+                          ]
```
[the remark this answers is not in the document]
Comment (plondon) [reply packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/MakeOffer/index.tsx:247]: Rather, a component that expands and collapses, when collapsed show only the total fee, when expanded show all fees.

File: packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/MakeOffer/index.tsx [hunk 4]
```diff
@@ -47,14 +72,28 @@ const MakeOffer: React.FC<Props> = (props) => {
...
+
+  const toggleTermsAccepted = () => {
+    setTermsAccepted(!termsAccepted)
+  }
+
+  const validate = (formValues: SendFormType, props: Props) => {
+    return {
+      amount: needsWrap && !canWrap ? 'NOT_ENOUGH_ETH' : true
+    }
+  }
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/MakeOffer/index.tsx:96]: nice

File: packages/blockchain-wallet-v4-frontend/src/components/Balances/selectors.ts [hunk 5]
```diff
@@ -107,6 +107,15 @@ export const getEthBalance = createDeepEqualSelector(
   }
 )
 
+export const getEthBalances = createDeepEqualSelector(
+  [getEthNonCustodialBalance, getCoinCustodialBalance('ETH')],
+  (balancesR, custodialBalanceR) => {
+    const custodialBalance = custodialBalanceR.getOrElse(0)
+
+    return Remote.of([new BigNumber(balancesR.getOrElse(new BigNumber(0))), custodialBalance])
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/components/Balances/selectors.ts:115]: Does it need to be a `BigNumber` inside the `getOrElse`?

Can it be just

```suggestion
    return Remote.of([new BigNumber(balancesR.getOrElse(0)), custodialBalance])
```

File: packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/Status/index.tsx [hunk 6]
```diff
@@ -1,9 +1,119 @@
...
 
-const NftOrderStatus: React.FC<Props> = (props) => {
-  return <div>{props.orderFlow.status}</div>
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding-top: 15em;
+  font-family: Inter, sans-serif;
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Nfts/NftOrder/Status/index.tsx:17]: If you use Text component you don't need to add font-family:

styled(Text)

File: packages/blockchain-wallet-v4-frontend/src/modals/Nfts/components/networkFees.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            <ChevronArea>
+              <Icon name='chevron-down' size='24px' color='grey400' />
+            </ChevronArea>
+          )}
+        </Top>
+        <Fees style={moreFees ? {} : { display: 'none' }}>
+          {/* @ts-ignore */}
+          <MakeOfferFees {...props} asset={val} />
+          {/* @ts-ignore */}
+          <WrapEthFees {...props} />
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/modals/Nfts/components/networkFees.tsx:93]: why the ts-ignores?

**Review by plondon — APPROVED** [review]

**Pull request merged**